sdi/transport.go

```diff
@@ -0,0 +1,76 @@
+package sdi
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httputil"
+)
+
+// Transport defines interface for http client transport methods
+type Transport interface {
+	SetTransport(t *http.Transport)
+	WithContext(ctx context.Context)
+	RoundTrip(r *http.Request) (*http.Response, error)
+}
+
+// DefaultTransport default configuration for HTTP communications
+type DefaultTransport struct {
+	HTTPTransport *http.Transport
+	Context       context.Context
```

Comment:
I wouldn't expect the context to be set inside something called "Transport". Context is based on the current execution context, not something that is long lived. As a general rule, the context should never be defined inside a struct as a variable, only passed on via method calls.

Reply:
TIL